sdk/storage/Azure.Storage.DataMovement/src/ConcurrencyTuner/ResourceMonitor.cs

```diff
@@ -134,6 +167,36 @@ internal async Task CalibrateComputation(CancellationToken cancellationToken)
             }
         }
 
+        public void StartMonitoring()
+        {
+            if (IsRunning)
+                return;
+
+            IsRunning = true;
```

Comment:
Should this be behind a lock / semaphoreslim ?

Reply:
Setting a boolean is atomic: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/variables#96-atomicity-of-variable-references

Also I'm not sure we'll be using this from multiple threads since there will only be one per transfer manager and I think the transfer manager will be the one starting/stopping it.

Reply:
Hm then that implementation of how transfer manager decides to start and stop it will be interesting, since it will have to juggle and determine from other jobs when to start and stop, when multiple jobs are running at the same time. But yes I agree probably a later problem